### PR TITLE
fivetran-destination: Use the provided Docker image

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR is centered around testing the Fivetran Destination, it does a few individual things:

1. Uses the Fivetran provided Docker image of the "destination tester" instead of building our own from our fork of the fivetran sdk.
2. Breaks up the existing "test_writes" into separate tests, since the testing behavior we were trying to do isn't yet supported (see [Slack](https://materializeinc.slack.com/archives/C060KAR4802/p1706548430680759)).
3. Updates our connector so we no longer splice the "_fivetran_synced" and "_fivetran_deleted" columns into various requests, since the tester now provides these.
4. Runs our Fivetran destination tests in CI.

There are a couple outstanding issues related to testing the Fivetran Destination that I have pinged them about and are currently in progress:

1. aarch64 based Docker image. Currently if you run these tests locally on a Mac it will run the x86 image in emulation mode. [Slack](https://materializeinc.slack.com/archives/C060KAR4802/p1706548430680759)
2. The "truncate" option in their testing framework is currently broken. [Slack](https://materializeinc.slack.com/archives/C060KAR4802/p1706651239233319).

Even with these issues I wanted to get this change up in case I switch to other work soon.

### Motivation

Fixes https://github.com/MaterializeInc/materialize/issues/24446

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/a

### CODE REVIEW

File: ci/test/pipeline.template.yml [hunk 1]
```diff
@@ -378,17 +378,32 @@ steps:
...
-    depends_on: build-aarch64
-    timeout_in_minutes: 40
-    inputs: [test/ssh-connection]
-    artifact_paths: junit_*.xml
-    plugins:
-      - ./ci/plugins/mzcompose:
-          composition: ssh-connection
-    agents:
-      queue: linux-aarch64
+  - group: "Connection tests"
```
Comment (ParkMyCar) [inline ci/test/pipeline.template.yml:381]: Totally up to the testing folks if we want to group these tests. It seemed like a decent idea but I'll 100% defer to y'all!

**Review by benesch — APPROVED** [review]
Very clean. Thanks for handling this!

File: ci/test/pipeline.template.yml (hunk 1 above)
Comment (def-) [reply ci/test/pipeline.template.yml:381]: Thanks, the grouping seems fine.

**Review by def- — APPROVED** [review]

**Review by jkosh44 — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
[Nightly](https://buildkite.com/materialize/nightlies/builds/6244) should be good, most failures would disappear after a rebase.

**Pull request merged**